src/renderer/dx/DxFontRenderData.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "../../renderer/inc/FontInfoDesired.hpp"
+#include "BoxDrawingEffect.h"
+
+#include <dwrite.h>
+#include <dwrite_1.h>
+#include <dwrite_2.h>
+#include <dwrite_3.h>
+
+#include <wrl.h>
+
+namespace Microsoft::Console::Render
+{
+    class DxFontRenderData
+    {
+    public:
+        struct LineMetrics
+        {
+            float gridlineWidth;
+            float underlineOffset;
+            float underlineOffset2;
+            float underlineWidth;
+            float strikethroughOffset;
+            float strikethroughWidth;
+        };
+
+        DxFontRenderData(::Microsoft::WRL::ComPtr<IDWriteFactory1> dwriteFactory) noexcept;
+
+        // DirectWrite text analyzer from the factory
+        [[nodiscard]] Microsoft::WRL::ComPtr<IDWriteTextAnalyzer1> Analyzer() noexcept;
+
+        [[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFallback> SystemFontFallback();
+
+        // A locale that can be used on construction of assorted DX objects that want to know one.
+        [[nodiscard]] std::wstring UserLocaleName();
+
+        [[nodiscard]] til::size GlyphCell() noexcept;
+        [[nodiscard]] LineMetrics GetLineMetrics() noexcept;

```

Comment:
I intended to use just `LineMetrics` as the method name, but there is a struct named `LineMetrics`, which confuses the compiler.

Reply:
I don't think it's worth figuring out the namespace collision so you can use the `LineMetrics` name. Just stick with the `Get*` one I suppose.